canary-checker/docs/reference/1-kubernetes-resource.mdx

```diff
@@ -0,0 +1,131 @@
+    },
+    {
+      field: 'static resources',
+      description:
+        'Static resources are like `resources` but preserved between checks. They are only deleted when the canary is deleted.',
+      scheme: '`[]KubernetesManifest`'
+    },
+    {
+      field: 'checks',
+      description:
+        'canary spec for the checks to be performed after the resources are created',
+      scheme: 'CanarySpec'
+    },
+    {
+      field: 'checkRetries',
+      description: 'Retry configuration for the checks',
+      scheme: '[CheckRetries](#check-retries)'
+    },
+    {
+      field: 'clearResources',
+      scheme: 'bool',
+      description: `When set to true, resources from previous checks will be deleted before every run. 
+        Even though the resources are deleted at the end of a check, setting this to \`true\` will guarantee that
+        there are no remains from a previous failed run.`
+    },
+    {
+      field: 'waitFor',
+      description:
+        'Specify the desired state of the static/non-static resources before running the checks',
+      scheme: '[WaitFor](#wait-for)'
+    }
+  ]}
+/>
+
+### Check Retries
+
+<Fields
+  rows={[
+    {
+      field: 'delay',
+      description: 'Initial delay before the checks are run',
+      scheme: 'Duration'
+    },
+    {
+      field: 'timeout',
+      description: 'Timeout for the check',
+      scheme: 'Duration'
+    },
+    {
+      field: 'interval',
+      description: 'Retry the checks, on failure, on this interval',
+      scheme: 'Duration'
+    }
+  ]}
+/>
+
+### Wait For
+
+<Fields
+  rows={[
+    {
+      field: 'expr',
+      description: `CEL expression that determines whether all the resources are in their desired state before running checks on them. 
+        It receives a \`resources\` array of the static and non-static resources.
+        The default behavior is to wait until all the resources are healthy \`dyn(resources).all(r, k8s.isHealthy(r))\`.`,
+      scheme: 'CEL'
+    },
+    {
+      field: 'disable',
+      description:
+        'Disable the default behavior of waiting for resources to be healthy.',
+      scheme: 'Duration'
+    },
+    {
+      field: 'delete',
+      description:
+        'When set to true, the check waits for the resources to be deleted',
+      scheme: 'bool'
+    },
+    {
+      field: 'timeout',
+      description:
+        'Timeout to wait for all static & non-static resources to satisfy the expression. _(Default: 10m)_',
+      scheme: 'Duration'
+    },
+    {
+      field: 'interval',
+      description:
+        'Interval to check if all static & non-static resources are ready. _(Default: 5s)_',
+      scheme: 'Duration'
+    }
+  ]}
+/>
+
+### Templating
+
+The static and non static resources can be templated. This is helpful to create resources with random names. Example:
+you can setup a resource to create a pod with random name on each check run. This way you don't have to wait for the pod to be deleted on every check.
```

Comment:
Can we create this example using a details section

Reply:
A fixture that showcases templating is in this PR https://github.com/flanksource/canary-checker/pull/1810